src/polus/plugins/_plugins/classes/plugin_methods.py

```diff
@@ -187,6 +213,83 @@ def __setattr__(self, name, value):
 
         super().__setattr__(name, value)
 
+    def _to_cwl(self):
+        """Return CWL yml as dict."""
+        cwl_dict = CWL_BASE_DICT
+        cwl_dict["inputs"] = {}
+        cwl_dict["outputs"] = {}
+        inputs = [input_to_cwl(x) for x in self.inputs]
+        inputs = inputs + [output_to_cwl(x) for x in self.outputs]
+        for inp in inputs:
+            cwl_dict["inputs"].update(inp)
+        outputs = [outputs_cwl(x) for x in self.outputs]
+        for out in outputs:
+            cwl_dict["outputs"].update(out)
+        cwl_dict["requirements"]["DockerRequirement"]["dockerPull"] = self.containerId
+        return cwl_dict
+
+    def save_cwl(self, path: typing.Union[str, pathlib.Path]):
+        """Save plugin as CWL command line tool."""
+        assert str(path).split(".")[-1] == "cwl", "Path must end in .cwl"
+        with open(path, "w") as file:
+            yaml.dump(self._to_cwl(), file)
+        return path
+
+    @property
+    def _cwl_io(self) -> dict:
+        """Dict of I/O for CWL."""
+        return {
+            x.name: io_to_yml(x) for x in self._io_keys.values() if x.value is not None
+        }
+
+    def save_cwl_io(self, path):
+        """Save plugin's I/O values to yml file to be used with CWL command line tool."""
+        self._check_inputs()
+        assert str(path).split(".")[-1] == "yml", "Path must end in .yml"
+        with open(path, "w") as file:
+            yaml.dump(self._cwl_io, file)
+        return path
+
+    def run_cwl(
+        self,
+        cwl_path: typing.Optional[typing.Union[str, pathlib.Path]] = None,
+        io_path: typing.Optional[typing.Union[str, pathlib.Path]] = None,
+    ):
+        """Run configured plugin in CWL.
+
+        Run plugin as a CWL command line tool after setting I/O values.
+        Two files will be generated: a CWL (`.cwl`) command line tool
+        and an I/O file (`.yml`). They will be generated in
+        current working directory if no paths are specified. Optional paths
+        for these files can be specified with arguments `cwl_path`,
+        and `io_path` respectively.
+
+        Args:
+            cwl_path: [Optional] target path for `.cwl` file
+            io_path: [Optional] target path for `.yml` file
+
+        """
+        if not self.outDir:
+            raise ValueError("")
+
+        if not cwl_path:
+            _p = pathlib.Path.cwd().joinpath(name_cleaner(self.name) + ".cwl")
+            _cwl = self.save_cwl(_p)
+        else:
+            _cwl = self.save_cwl(cwl_path)
+
+        if not io_path:
+            _p = pathlib.Path.cwd().joinpath(name_cleaner(self.name) + ".yml")
+            self.save_cwl_io(_p)  # saves io to make it visible to user
+        else:
+            self.save_cwl_io(io_path)  # saves io to make it visible to user
+
+        outdir_path = relpath(self.outDir.parent)  # type: ignore
```

Comment:
Unfortunately, this way of invoking the cwltool python API does not allow passing several important command line flags. I recommend calling `cwltool.main.main` together with the --write-summary flag.

See https://github.com/PolusAI/workflow-inference-compiler/blob/master/src/wic/run_local.py#L133